src/v/cluster/rm_stm.cc

```diff
@@ -2634,4 +2635,19 @@ ss::future<> rm_stm::clear_old_tx_pids() {
       });
 }
 
+ss::future<> rm_stm::clear_old_idempotent_pids() {
+    while (_log_state.idempotent_pids.size() > _max_saved_pids_count()) {
+        auto pids_it = _log_state.idempotent_pids.begin();
+        time_point_type key_for_delete
+          = _log_state.idempotent_pids.begin()->first;
+        while (pids_it->first == key_for_delete) {
+            _log_state.seq_table.erase(pids_it->second);
+            _inflight_requests.erase(pids_it->second);
+            _log_state.idempotent_pids.erase(pids_it++);
+        }
```

Comment:
i do not understand this part, it is a map, we expect it to have unique keys, why do wee need to loop until the key is different ?

Reply:
it is multimap. From time to pids

Reply:
got it, didn't see it is a multi map sorry.